### PR TITLE
Dev/backbone

### PR DESCRIPTION
**Pull request by JLYang**
Pls look at this 

- backbone 
- main 
- integration (slight changes to dataset)

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -47,10 +47,14 @@ def __len__(self):
...
+        #      "Embedded": self.embedded_docs[idx, :, :]}
+        # y = {"Politeness Score": self.politeness_scores[idx],
+        #      "Length": self.lens[idx]}
+        # unembed = self.unembedded_docs[idx]
+        embed = self.embedded_docs[idx, :, :]
+        score = self.politeness_scores[idx]
+        doc_len = self.lens[idx]
 
-        return x, y
+        return embed, score, doc_len
```
Comment (ryrodrig112) [inline src/utils.py:59]: Will it work to return more than x, y for DataLoader? I was trying to follow the template from [here](https://pytorch.org/tutorials/beginner/basics/data_tutorial.html#:~:text=__getitem__,-The%20__getitem&text=Based%20on%20the%20index%2C%20it,corresponding%20label%20in%20a%20tuple.) -- if it works all good, just wanna confirm

**Review by ryrodrig112 — COMMENTED** [review]
Left a comment on return

**Comment by ryrodrig112** [comment]
Mostly looks good - feel free to merge after looking at the comment I'd left

File: src/utils.py (hunk 1 above)
Comment (JLYang) [reply src/utils.py:59]: Yep it works. Doesn't have to be just x, y. DataLoader just takes the output of get item and stacks batch_size number of them (that's why I took out the unembedded version - variable size can't be stacked).

**Pull request merged**